github/prci_github/internals.py

```diff
@@ -211,14 +216,23 @@ def take(self, runner_id):
 
     def execute(self, exc_handler=None):
         depends_results = {}
-        for dep in self.requires:
-            status = Status(self.repo, self.pull, dep)
-            depends_results[dep] = JobResult(
-                status.state, status.description, status.target_url)
+
+        # sometimes, when running runners in parallel, it may happen
+        # that we get an exception from OpenSLL.
+        @retry(OpenSSL.SSL.Error)
```

Comment:
How does this work? It is reasonable to assume if it fails once, it's going to pass the next time around?

Reply:
The `retry` decorator will re-execute the method if the exception given as parameter is raised. 
In my many tests, it only happened to fail once, but as you said it could fail more times. We could change `retry` to receive the number of attempts that it should do. Something like: `@retry(OpenSSL.SSL.Error, attempts=2)`.

TBH I don't like the approach of having the `retry` decorator, however, given the deadlines, I think it's easier to go with it than change the code to use `threading`.